dmgbuild/core.py

```diff
@@ -129,7 +129,7 @@ def load_json(filename, settings):
     settings['icon_locations'] = icon_locations
 
 def build_dmg(filename, volume_name, settings_file=None, settings={},
-              defines={}, lookForHiDPI=True):
+              defines={}, lookForHiDPI=True, detach_timeout=5):
```

Comment:
I don't like calling this `detach_timeout`, because it isn't a timeout. It's a retry count. I think it's otherwise a good idea, so if you change it to `detach_retries` I'll merge the change.